Tests are failing in `develop` but pass in CI

# Problem

```
=== RUN   TestRuntimeLogLevelIsNotWrittenToDatabase
INFO [06-21|10:49:33.838|github.com/status-im/status-go/api/geth_backend.go:108] Status backend initialized               backend=geth version= commit= IpfsGatewayURL=
WARN [06-21|10:49:33.904|github.com/status-im/status-go/appdatabase/database.go:123] Migrating accounts: no wallet_root_address found in settings 
    backend_test.go:747: 
        	Error Trace:	/Users/igorsirotin/Repositories/Status/status-go/api/backend_test.go:747
        	Error:      	Received unexpected error:
        	            	sql: expected 24 destination arguments in Scan, not 25
        	Test:       	TestRuntimeLogLevelIsNotWrittenToDatabase
--- FAIL: TestRuntimeLogLevelIsNotWrittenToDatabase (0.15s)
```

And the CI is green although 3 reruns fails.

<img width="1027" alt="image" src="https://github.com/status-im/status-go/assets/25482501/32df2cf2-75e3-40ae-bc9b-44f9d7e182ae">

in :
- https://github.com/status-im/status-go/pull/5362
- nightly: https://ci.status.im/job/status-go/job/tests-nightly/258/testReport/
- latest `develop`

--- a/_assets/scripts/run_unit_tests.sh
+++ b/_assets/scripts/run_unit_tests.sh
@@ -88,8 +88,10 @@ run_test_for_package() {
     -v ${GOTEST_EXTRAFLAGS} \
     -timeout "${package_timeout}" \
     -count 1 \
-    -tags "${BUILD_TAGS}" |
+    -tags "${BUILD_TAGS}" | \
     redirect_stdout "${output_file}"
+
+  local go_test_exit=$?
 
   # Merge package coverage results
   go run ./cmd/test-coverage-utils/gocovmerge.go ${package_dir}/coverage.out.rerun.* > ${coverage_file}
@@ -97,7 +99,6 @@ run_test_for_package() {
   # Cleanup coverage reports
   rm -f ${package_dir}/coverage.out.rerun.*
 
-  local go_test_exit=$?
   echo "${go_test_exit}" > "${exit_code_file}"
   if [[ "${go_test_exit}" -ne 0 ]]; then
     if [[ "${CI}" == 'true' ]]; then

